test/e2e/utils.go

```diff
@@ -576,6 +582,70 @@ func wasUsingOtelCollector(jaegerInstanceName, namespace string) bool {
 	return false
 }
 
+func verifyAllInOneImage(jaegerInstanceName, namespace string, expected bool) {
+	require.Equal(t, expected, wasUsingOtelAllInOne(jaegerInstanceName, namespace))
+}
+
+func wasUsingOtelAllInOne(jaegerInstanceName, namespace string) bool {
+	deployment, err := fw.KubeClient.AppsV1().Deployments(namespace).Get(context.Background(), jaegerInstanceName, metav1.GetOptions{})
+	require.NoError(t, err)
+	containers := deployment.Spec.Template.Spec.Containers
+	for _, container := range containers {
+		if container.Name == "jaeger" {
+			logrus.Infof("Test %s is using image %s", t.Name(), container.Image)
+			return strings.Contains(container.Image, "opentelemetry-all-in-one")
+		}
+	}
+
+	return false
+}
+
+func verifyAgentImage(appName, namespace string, expected bool) {
+	require.Equal(t, expected, wasUsingOtelAgent(appName, namespace))
+}
+
+// Was this Jaeger Instance using the OTEL agent?
+func wasUsingOtelAgent(appName, namespace string) bool {
+	var pods *corev1.PodList
+	var pod corev1.Pod
+
+	// Sometimes the app gets redeployed twice and we can get three pods, wait till there are either 1 or 2
+	err := wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		pods, err = fw.KubeClient.CoreV1().Pods(namespace).List(context.Background(), metav1.ListOptions{LabelSelector: "app=" + appName})
+		require.NoError(t, err)
+		if len(pods.Items) > 0 && len(pods.Items) < 3 {
+			return true, nil
+		}
+		return false, nil
+	})
+	require.NoError(t, err)
+
+	if len(pods.Items) == 1 {
+		logrus.Infof("Found 1 pod %s", pods.Items[0].Name)
+		pod = pods.Items[0]
+	} else {
+		for _, p := range pods.Items {
+			if p.DeletionTimestamp == nil {
```

Comment:
is there a way to query pods that are not scheduled for deletion (if I got correctly what `DeletionTimestamp` means)?

Reply:
You're correct, that's what we're looking for.  I don't know any way to do this.  @jpkrohling  do you?

Reply:
I don't really know, perhaps there's something in the API docs about this?